mllib/src/main/scala/org/apache/spark/ml/param/params.scala

```diff
@@ -379,7 +427,6 @@ trait Params extends Identifiable with Serializable {
    * @param value  the default value
    */
   protected final def setDefault[T](param: Param[T], value: T): this.type = {
-    shouldOwn(param)
```

Comment:
This causes some racing condition. Because not all members are initialized before `setDefault` is called. `shouldOwn` checks `params`, which is a lazy val. If `params` is initialized before some params, it would contains `null`. I didn't figure out a good solution. Given that this is not called by users, I removed this check and left a note on `params`.